addons/event-audit/common/src/main/java/org/commonjava/indy/event/audit/EventAuditListener.java

```diff
@@ -116,24 +126,37 @@ private void transformFileEvent( org.commonjava.maven.galley.event.FileEvent eve
             return;
         }
 
-        final String path = transfer.getPath();
-
-        fileEvent.setTargetPath( path );
-        //TODO figure out what's the NodeId
-        fileEvent.setNodeId( "" );
-        fileEvent.setSessionId( trackingKey.getId() );
-        fileEvent.setTimestamp( new Date() );
-
-        final KeyedLocation keyedLocation = (KeyedLocation) transfer.getLocation();
+        final Location location = transfer.getLocation();
+        if ( !( location instanceof KeyedLocation ) )
+        {
+            logger.trace( "Not in a keyed location: {}", transfer );
+            return;
+        }
 
         try
         {
-            StoreKey affectedStore = keyedLocation.getKey();
-            final Transfer txfr = downloadManager.getStorageReference( keyedLocation.getKey(), path );
-            if ( txfr.exists() )
+            final KeyedLocation keyedLocation = (KeyedLocation) location;
+            final StoreKey affectedStore = keyedLocation.getKey();
+            if ( affectedStore.getType() == group )
+            {
+                logger.trace( "Not auditing content stored directly in group: {}. This content is generally aggregated metadata, and can be recalculated. Groups may not be stable in some build environments",
+                              affectedStore );
+                return;
+            }
+
+            final String path = transfer.getPath();
+
+            fileEvent.setTargetPath( path );
+            //TODO figure out what's the NodeId
+            fileEvent.setNodeId( "" );
+            fileEvent.setSessionId( trackingKey.getId() );
+            //fileEvent.setTimestamp( new Date() );
```

Comment:
Not sure why this line caused the ftest failed, checking ...
`mvn -Prun-its -DskipSurefire=true clean install -Dit.test=org.commonjava.indy.folo.ftest.report.RecalculateMissingChecksumForTrackingRecordTest`